src/components/Header.tsx

```diff
@@ -23,14 +23,9 @@ const Header: StatelessComponent<HeaderProps> = ({ handleSidebarToggle }): React
       const ver = e.target.value;
     
       if (currentNode) {
-        const newPath = currentNode.slug.replace(/^\/en\/[0-9]+\//, `/en/${ver}/`);
-        const otherNode = getNodes().find(n => n.slug === newPath);
-        // This has to be a hard refresh, because the sidebar needs to unmount
-        if (otherNode) {
-          navigate(otherNode.slug);
-        } else {
-          navigate(`/en/${ver}`);
-        }
+        const newPath = getVersionPath(currentNode, ver);
+        // This has to be a hard refresh, because the sidebar and searchbar need to unmount
+        navigate(newPath);
```

Comment:
This ultimately still does the same thing it was doing before - it's just a lot cleaner.